spec/unit/lib/cloud_controller/dependency_locator_spec.rb

```diff
@@ -484,6 +484,41 @@
 
       expect(locator.statsd_client).to eq(expected_client)
     end
+
+    it 'returns the null statsd client' do
+      host = 'test-host'
+      port = 1234
+
+      TestConfig.override(
+        statsd_host: host,
+        statsd_port: port,
+        enable_statsd_metrics: false
+      )
+
+      expected_client = double(CloudController::NullStatsdClient)
+
+      allow(CloudController::NullStatsdClient).to receive(:new).
+        and_return(expected_client)
+
+      expect(locator.statsd_client).to eq(expected_client)
+    end
+
+    it 'returns the statsd client for other than api vms' do
+      host = 'test-host'
+      port = 1234
+      TestConfig.context = :deployment_updater
+      TestConfig.override(
```

Comment:
This takes the default values from the `cloud_controller.yml`, i.e. `enable_statsd_metrics: true`. I suggest that you overwrite it with `nil` in this test case.